internal/configsource/fileconfigsource/README.md

```diff
@@ -0,0 +1,42 @@
+# File Config Source (Alpha)
+
+Use the file config source to inject YAML fragments or scalars into the
+configuration.
+
+## Configuration
+
+Under the `config_sources:` use `file:` or `file/<name>:` to create a
+file config source. The following parameters are available to
+customize file config sources:
```

Comment:
I know this is used elsewhere, but some specificity and removing parameter boilerplate may be helpful (since there is only the optional name):

```suggestion
Under the top-level `config_sources:` component mapping use `file:` or `file/<component_name>:` to create a
file config source capable of reading from subsequently specified files and detecting their changes:
```

```suggestion
Under the `config_sources:` use `file:` or `file/<name>:` to create a
file config source. The following parameters are available to
customize file config sources:
```